### PR TITLE
[DEV-2394] Change discount code to special code

### PR DESCRIPTION
**Pull request by Kolibri1990**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Review by davidleomay — APPROVED** [review]
- [ ] Wait for approval from @Yannick1712

File: src/shared/services/payment-info.service.ts [hunk 1]
```diff
@@ -40,6 +40,10 @@ export class PaymentInfoService {
         throw new BadRequestException(`Asset not buyable ${'paymentMethod' in dto ? 'via Bank' : ''}`);
     }
 
+    if ('discountCode' in dto) {
+      dto.specialCode = dto.discountCode;
+    }
```
Comment (Yannick1712) [inline src/shared/services/payment-info.service.ts:45]: Don't need the `{}`

File: src/shared/services/payment-info.service.ts [hunk 2]
```diff
@@ -67,6 +71,10 @@ export class PaymentInfoService {
         'CHF transactions are only permitted to Liechtenstein or Switzerland. Use EUR for other countries.',
       );
 
+    if ('discountCode' in dto) {
+      dto.specialCode = dto.discountCode;
+    }
+
```
Comment (Yannick1712) [inline src/shared/services/payment-info.service.ts:77]: Same

File: src/shared/services/payment-info.service.ts [hunk 3]
```diff
@@ -93,6 +101,10 @@ export class PaymentInfoService {
     if (jwt && !dto.targetAsset.isBuyableOn(jwt.blockchains))
       throw new BadRequestException('Asset blockchain mismatch');
 
+    if ('discountCode' in dto) {
+      dto.specialCode = dto.discountCode;
+    }
+
```
Comment (Yannick1712) [inline src/shared/services/payment-info.service.ts:107]: Same

**Review by Yannick1712 — APPROVED** [review]

**Review by davidleomay — APPROVED** [review]

**Pull request merged**